eo-parser/src/test/resources/org/eolang/parser/packs/optimize/redundant-levels-with-arrays.yaml

```diff
@@ -0,0 +1,70 @@
+xsls:
+  - /org/eolang/parser/wrap-method-calls.xsl
+  - /org/eolang/parser/synthetic-references.xsl
+  - /org/eolang/parser/optimize/abstracts-float-up.xsl
+  - /org/eolang/parser/optimize/remove-levels.xsl
+tests:
+  - /program/errors[count(*)=0]
+  # 'main' object
+  - //o[@name='main']
+  - //o[@name='main']/o[@base='tuple' and @name='arr' and count(o)=2]/o[@base='int']
+  - //o[@name='main']/o[@base='tuple' and @name='arr' and count(o)=2]/o[not(@base='int')]
+  - //o[@name='main']/o[@base='eq' and @name='@' and count(o)=2]
+  - //o[@name='main']/o[@base='eq' and @name='@' and count(o)=2]/o[@base='tuple']/o/o[@base='arr']
+  - //o[@name='main']/o[@base='eq' and @name='@' and count(o)=2]/o[@base='tuple']/o[@base='int']
+  # The first generated object: 'constant-array' from the example below
+  - //o[@abstract and count(o)=2 and @name]
+  - //o[@abstract and count(o)=2 and @name]/o[@base='tuple' and count(o)=2]
+  - //o[@abstract and count(o)=2 and @name]/o[@base='tuple' and count(o)=2]/o[@base='int' and contains(text(), '1')]
+  - //o[@abstract and count(o)=2 and @name]/o[@base='tuple' and count(o)=2]/o[@base='int' and contains(text(), '2')]
+  - //o[@abstract and count(o)=2 and @name]/o[@name='@']
+  # The second generated object: 'second' from the example below
+  - //o[@abstract and count(o)=3 and @name]
+  - //o[@abstract and count(o)=3 and @name]/o[@base='.at' and count(o)=2]
+  - //o[@abstract and count(o)=3 and @name]/o[@base='.at' and count(o)=2]/o/o[@base='arr']
+  - //o[@abstract and count(o)=3 and @name]/o[@base='.at' and count(o)=2]/o[@base='int' and contains(text(), '1')]
+  - //o[@abstract and count(o)=3 and @name]/o[@name='@']
+  # The third generated object: 'first' from the example below
+  - //o[@abstract and count(o)=3 and @name]
+  - //o[@abstract and count(o)=3 and @name]/o[@base='.at' and count(o)=2]
+  - //o[@abstract and count(o)=3 and @name]/o[@base='.at' and count(o)=2]/o[@base='arr']
+  - //o[@abstract and count(o)=3 and @name]/o[@base='.at' and count(o)=2]/o[@base='int' and contains(text(), '1')]
+  - //o[@abstract and count(o)=3 and @name]/o[@name='@']
+# Currently the test converts the code from the snippet to:
+# ____
+# [] > main
+#   tuple > arr
+#     0
+#     constant-array
+#   eq > @
+#     tuple
+#       second
+#         arr
+#     tuple
+#       2
+#
+# [] > constant-array
+#   tuple > local-element
+#     1
+#     2
+#   local-element > @
+#
+# [arr] > second
+#   .at > local-element
+#     first
+#       arr
+#     1
+#   local-element > @
+#
+# [arr] > first
+#   .at > local-element
+#     arr
+#     1
+#   local-element > @
+# ____
+eo: |
+  [] > main
+    * 0 (* 1 2) > arr
+    eq > @
+      * ((arr.at 1).at 1)
+      * 2
```

Comment:
@volodya-lombrozo As far as I know this object should be just `2`. Not a `tuple`.  Since `((arr.at 1).at 1)` returns the second element of the tuple which is `2`. Maybe I'm wrong. What do you think?

Reply:
@Graur You are right:
1) `((arr.at 1).at 1` returns `2`
___
2) So, we actually compare `* 2` and `* 2`
3) Why not just `2` and `2`? Because in this case `((arr.at 1).at 1)` will contain redundant parentheses (it's an error).
4) If I will use just a `(arr.at 1).at 1` (without parenthesis), then test won't catch the problem with scopes.

Reply:
@Graur Please, ask else if something is unclear.